model/interactive/src/main/proto/beam_interactive_api.proto

```
@@ -0,0 +1,160 @@
+ * the TestStream class in order to control a pipeline remotely.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.model.interactive.v1;
+
+option go_package = "interactive_v1";
+option java_package = "org.apache.beam.model.interactive.v1";
+option java_outer_classname = "BeamInteractiveApi";
+
+import "beam_runner_api.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+
+service InteractiveService {
+  // A TestStream will request for events using this RPC.
+  rpc Events(EventsRequest) returns (stream EventsResponse) {}
+
+  // Starts the stream of events to the EventsRequest.
+  rpc Start (StartRequest) returns (StartResponse) {}
+
+  // Advances the stream to the specified offset then pauses the stream. This
+  // starts the stream if it is not RUNNING.
+  rpc Advance (AdvanceRequest) returns (AdvanceResponse) {}
+
+  // Stops and resets the stream to the beginning.
+  rpc Stop (StopRequest) returns (StopResponse) {}
+
+  // Pauses the stream of events to the EventsRequest. If there is already an
+  // outstanding EventsRequest streaming events, then the stream will pause
+  // after the EventsResponse is completed.
+  // To un-pause, send either a Start or Advance request.
+  rpc Pause (PauseRequest) returns (PauseResponse) {}
+
+  // Sends a single element to the EventsRequest then closes the stream.
+  rpc Step (StepRequest) returns (StepResponse) {}
+
+  // Responds with debugging and other cache-specific metadata.
+  rpc Status (StatusRequest) returns (StatusResponse) {}
+}
+
+message StartRequest {
+  // (Optional) How quickly the stream will be played back, e.g. if
+  // playback_speed == 2, then the stream will replay events twice as fast as
+  // they were recorded. If unspecified, this will default to 1.
+  double playback_speed = 1;
+
+  // (Optional) if present, will start the stream at the specified timestamp.
+  google.protobuf.Timestamp start_at = 2;
```

I'm not sure whether we should allow this feature since that means re-execution with start_at specified means different input data and it kinda defeats the purpose of replaying. Also, if we use Start method for also unpausing and changing playback speed, start_at either should be ignored or should return an error.

---

Good point, removed it.